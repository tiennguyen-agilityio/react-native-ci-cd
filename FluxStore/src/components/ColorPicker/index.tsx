import {memo} from 'react';

// Stores
import {useThemeStore} from '@/stores';

// Components
import Text from '../Text';
import Flex from '../Flex';
import Dot from '../Dot';

const DOT_SIZE = 34;
interface ColorPickerProps {
  colors: string[];
  size?: number;
  onValueChange: (value: string) => void;
  defaultValue?: string;
}

const ColorPicker = ({
  colors,
  defaultValue = '',
  size = DOT_SIZE,
  onValueChange,
}: ColorPickerProps) => {
  const {
    theme: {text},
  } = useThemeStore();

  return (
    <Flex>
      <Text style={{color: text.tertiary}}>Color</Text>
      <Flex marginTop={10} direction="row" gap={8}>
        {colors.map((color, index) => {
          const isSelected = defaultValue === color;
          const handleToggleSelect = () => {
            if (color !== defaultValue) {
              onValueChange(color);
            }
          };

          return (
            <Dot
              key={index}
              color={color}
              size={size}
              hasBorder={isSelected}
              onSelect={handleToggleSelect}
            />
          );
        })}
      </Flex>
    </Flex>
  );
};

export default memo(ColorPicker);
