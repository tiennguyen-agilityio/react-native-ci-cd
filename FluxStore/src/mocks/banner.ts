export const BANNERS = [
  {
    id: '01',
    title: 'Clothing Collection 2025',
    imageUrl: 'https://i.ibb.co/wZv2s1T6/Frame-33149.png',
  },
  {
    id: '02',
    title: 'Autumn Collection 2025',
    imageUrl: 'https://i.ibb.co/hRBFHvHk/Mask-Group-1.png',
  },
  {
    id: '03',
    title: 'Accessories Collection 2025',
    imageUrl: 'https://i.ibb.co/20WCrpXW/Frame-33123.png',
  },
];
