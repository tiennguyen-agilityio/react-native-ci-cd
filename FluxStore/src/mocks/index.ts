export * from './product';
export * from './banner';
export * from './review';
export * from './cart';
