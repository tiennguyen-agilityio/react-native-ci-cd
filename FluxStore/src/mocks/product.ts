import {Product} from '@/interfaces';

export const COLORS = ['black', 'red', 'blue'];
export const SIZES = ['S', 'M', 'L'];

export const PRODUCTS: Product[] = [
  {
    id: '01',
    name: 'Linen Dress',
    price: 90.02,
    discount: 10,
    rating: 5,
    reviewCount: 20,
    image: 'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
    sizes: ['s', 'M'],
    colors: ['black', 'red', 'blue'],
    carouselImages: [
      'https://i.ibb.co/KQj6DyL/image-87.png',
      'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
      'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    ],
  },
  {
    id: '02',
    name: 'Filted Waist Dress',
    price: 82.0,
    discount: 0,
    rating: 3,
    reviewCount: 110,
    image: 'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
    sizes: ['XL'],
    colors: ['pink', 'black'],
    carouselImages: [
      'https://i.ibb.co/KQj6DyL/image-87.png',
      'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
      'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    ],
  },
];
