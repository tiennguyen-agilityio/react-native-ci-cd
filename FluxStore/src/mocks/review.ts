export const REVIEWS = [
  {
    id: '01',
    avatarUrl: 'https://i.ibb.co/h1n0FRHn/Ellipse-76.png',
    name: 'Jennifer Rose',
    rating: 5,
    comment:
      'I love it.  Awesome customer service!! Helped me out with adding an additional item to my order. Thanks again!',
    createdAt: 1747710042553,
  },
  {
    id: '02',
    avatarUrl: 'https://i.ibb.co/tT8QbLVp/Ellipse-77.png',
    name: 'Kelly Rihana',
    rating: 5,
    comment: "I'm very happy with order, It was delivered on and good quality. Recommended!",
    createdAt: 1747710042570,
  },
];
