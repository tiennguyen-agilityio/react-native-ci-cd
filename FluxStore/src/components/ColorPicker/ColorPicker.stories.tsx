import {Meta, StoryObj} from '@storybook/react';

import ColorPicker from '.';
import {COLORS} from '@/mocks';
import {Alert} from 'react-native';

const meta: Meta<typeof ColorPicker> = {
  title: 'Components/ColorPicker',
  component: ColorPicker,
};

export default meta;

type Story = StoryObj<typeof ColorPicker>;

const defaultProps = {
  colors: COLORS,
  defaultValue: COLORS[0],
  onValueChange: (color: string) => Alert.alert('Current colors', color),
};

export const Default: Story = {
  args: defaultProps,
};
