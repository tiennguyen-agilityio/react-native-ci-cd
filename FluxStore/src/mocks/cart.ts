import {PRODUCTS} from './product';

export const CARTS = [
  {
    id: '01',
    product: {
      id: PRODUCTS[0].id,
      image: PRODUCTS[0].image,
      name: PRODUCTS[0].name,
      price: PRODUCTS[0].price,
    },
    sizes: [PRODUCTS[0]?.sizes?.[0], PRODUCTS[0]?.sizes?.[1]].join(', '),
    colors: [PRODUCTS[0]?.colors?.[0]].join(', '),
    quantity: 2,
    isChecked: true,
  },
  {
    id: '02',
    product: {
      id: PRODUCTS[1].id,
      image: PRODUCTS[1].image,
      name: PRODUCTS[1].name,
      price: PRODUCTS[1].price,
    },
    sizes: [PRODUCTS[1]?.sizes?.[0]].join(', '),
    colors: [PRODUCTS[1]?.colors?.[0], PRODUCTS[1]?.colors?.[1]].join(', '),
    quantity: 1,
    isChecked: false,
  },
];
