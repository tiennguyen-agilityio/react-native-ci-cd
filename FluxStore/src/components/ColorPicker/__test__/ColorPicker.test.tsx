import {render} from '@testing-library/react-native';

import {COLORS} from '@/mocks';
import ColorPicker from '..';

describe('ColorPicker Component', () => {
  const onValueChange = jest.fn();
  const props = {
    colors: COLORS,
    onValueChange,
  };
  it('renders correctly', () => {
    const {toJSON} = render(<ColorPicker {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with defaultIndex 1', () => {
    const {toJSON} = render(<ColorPicker {...props} defaultValue={COLORS[0]} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
