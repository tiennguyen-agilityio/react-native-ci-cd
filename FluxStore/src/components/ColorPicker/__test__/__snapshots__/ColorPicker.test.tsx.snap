// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Component renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "alignItems": "stretch",
      },
      {
        "flexWrap": "nowrap",
      },
      undefined,
      {},
    ]
  }
  testID="flex"
>
  <Text
    style={
      [
        {
          "letterSpacing": 0,
          "textAlign": "left",
          "textAlignVertical": "center",
        },
        {
          "color": "#737680",
          "fontFamily": "ProductSansRegular",
          "fontSize": 14,
        },
        {},
        {
          "color": "#777e90",
        },
      ]
    }
  >
    Color
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "alignItems": "stretch",
        },
        {
          "flexWrap": "nowrap",
        },
        undefined,
        {
          "gap": 8,
          "marginTop": 10,
        },
      ]
    }
    testID="flex"
  >
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "height": 34,
            "width": 34,
          },
        ]
      }
      testID="flex"
    >
      <View
        style={
          [
            {
              "backgroundColor": "black",
              "borderRadius": "50%",
              "height": 24,
              "shadowColor": "#000",
              "width": 24,
            },
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            },
          ]
        }
        testID="dot-black"
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "height": 34,
            "width": 34,
          },
        ]
      }
      testID="flex"
    >
      <View
        style={
          [
            {
              "backgroundColor": "red",
              "borderRadius": "50%",
              "height": 24,
              "shadowColor": "#000",
              "width": 24,
            },
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            },
          ]
        }
        testID="dot-red"
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "height": 34,
            "width": 34,
          },
        ]
      }
      testID="flex"
    >
      <View
        style={
          [
            {
              "backgroundColor": "blue",
              "borderRadius": "50%",
              "height": 24,
              "shadowColor": "#000",
              "width": 24,
            },
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            },
          ]
        }
        testID="dot-blue"
      />
    </View>
  </View>
</View>
`;

exports[`ColorPicker Component renders correctly with defaultIndex 1 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "alignItems": "stretch",
      },
      {
        "flexWrap": "nowrap",
      },
      undefined,
      {},
    ]
  }
  testID="flex"
>
  <Text
    style={
      [
        {
          "letterSpacing": 0,
          "textAlign": "left",
          "textAlignVertical": "center",
        },
        {
          "color": "#737680",
          "fontFamily": "ProductSansRegular",
          "fontSize": 14,
        },
        {},
        {
          "color": "#777e90",
        },
      ]
    }
  >
    Color
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "alignItems": "stretch",
        },
        {
          "flexWrap": "nowrap",
        },
        undefined,
        {
          "gap": 8,
          "marginTop": 10,
        },
      ]
    }
    testID="flex"
  >
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "height": 34,
            "width": 34,
          },
        ]
      }
      testID="flex"
    >
      <View
        style={
          [
            {
              "backgroundColor": "black",
              "borderColor": "#fff",
              "borderRadius": "50%",
              "borderWidth": 5,
              "elevation": 4,
              "height": 34,
              "shadowOffset": {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 6,
              "width": 34,
            },
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            },
          ]
        }
        testID="dot-black"
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "height": 34,
            "width": 34,
          },
        ]
      }
      testID="flex"
    >
      <View
        style={
          [
            {
              "backgroundColor": "red",
              "borderRadius": "50%",
              "height": 24,
              "shadowColor": "#000",
              "width": 24,
            },
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            },
          ]
        }
        testID="dot-red"
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "height": 34,
            "width": 34,
          },
        ]
      }
      testID="flex"
    >
      <View
        style={
          [
            {
              "backgroundColor": "blue",
              "borderRadius": "50%",
              "height": 24,
              "shadowColor": "#000",
              "width": 24,
            },
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            },
          ]
        }
        testID="dot-blue"
      />
    </View>
  </View>
</View>
`;
